### ISSUE TITLE
vswhere for minimal MSVC installations

### ISSUE THREAD

**Issue by jmcker** [cmake-js/cmake-js#183]
The `vswhere` technique created by @BurningEnlightenment and @nik-m2 in #178 and #180 can be expanded to include partial or minimal installs of Visual Studio's toolchain.

I can personally verify that the `vswhere` command included in #178 detects my minimal installation of MSVC for both VS2017 and VS2019 (I've included the output below).

I setup my Windows environment with Chocolatey:

```powershell
choco install visualstudio2017-workload-vctools
choco install visualstudio2019-workload-vctools
```

To expand the use case, the Nodejs 12.6.0 installer handles things for `node-gyp` in a similar way:
```powershell
# From C:\Program Files\nodejs\install_tools.bat:
choco upgrade -y python2 visualstudio2017-workload-vctools
```
![Node-setup](https://user-images.githubusercontent.com/25001741/60752186-2d8b8e00-9f88-11e9-8c6f-13fcab8424c3.png)

The `windows-build-tools` package seems to be doing something similar behind the scenes as well. I can't confirm this for sure, but it **IS** detected by the `vswhere` query.

Detection of `Microsoft.VisualStudio.Component.VC.Tools.x86.x64` seems to cover most/all cases. Even if it were to leave something out, detecting a majority would be a great improvement. The current behavior can be especially confusing since `cmake` has no problem detecting and handling these installs on its own.

As a workaround right now, I use the `npm config` variables introduced in [#100](https://github.com/cmake-js/cmake-js/issues/100#issuecomment-457218046) to hardcode my generator, but I would love to see this auto-detection grow from the changes in #180.
```powershell
npm config set cmake_js_G 'Visual Studio 15 2017 Win64'
```

`vswhere` Output:
---------------

Only `visualstudio2019-workload-vctools`:
```
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationVersion
16.1.29025.244
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
Visual Studio Locator version 2.6.7+91f4c1d09e [query version 2.1.1046.44959]
Copyright (C) Microsoft Corporation. All rights reserved.

instanceId: 0ecf9214
installDate: 7/5/2019 6:02:28 PM
installationName: VisualStudio/16.1.5+29025.244
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
installationVersion: 16.1.29025.244
productId: Microsoft.VisualStudio.Product.BuildTools
productPath: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\LaunchDevCmd.bat
state: 4294967295
isComplete: 1
isLaunchable: 1
isPrerelease: 0
isRebootRequired: 0
displayName: Visual Studio Build Tools 2019
description: The Visual Studio Build Tools allows you to build native and managed MSBuild-based applications without requiring the Visual Studio IDE. There are options to install the Visual C++ compilers and libraries, MFC, ATL, and C++/CLI support.
channelId: VisualStudio.16.Release
channelUri: https://aka.ms/vs/16/release/channel
enginePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service
releaseNotes: https://go.microsoft.com/fwlink/?LinkId=660893#16.1.5
thirdPartyNotices: https://go.microsoft.com/fwlink/?LinkId=660909
updateDate: 2019-07-05T23:02:28.6552697Z
catalog_buildBranch: d16.1
catalog_buildVersion: 16.1.29025.244
catalog_id: VisualStudio/16.1.5+29025.244
catalog_localBuild: build-lab
catalog_manifestName: VisualStudio
catalog_manifestType: installer
catalog_productDisplayVersion: 16.1.5
catalog_productLine: Dev16
catalog_productLineVersion: 2019
catalog_productMilestone: RTW
catalog_productMilestoneIsPreRelease: False
catalog_productName: Visual Studio
catalog_productPatchVersion: 5
catalog_productPreReleaseMilestoneSuffix: 1.0
catalog_productSemanticVersion: 16.1.5+29025.244
catalog_requiredEngineVersion: 2.1.3096.46871
properties_campaignId:
properties_channelManifestId: VisualStudio.16.Release/16.1.5+29025.244
properties_nickname:
properties_setupEngineFilePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installershell.exe
```

Both `visualstudio2019-workload-vctools` and `visualstudio2017-workload-vctools`:
```
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationVersion
15.9.28307.718
16.1.29025.244
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
Visual Studio Locator version 2.6.7+91f4c1d09e [query version 2.1.1046.44959]
Copyright (C) Microsoft Corporation. All rights reserved.

instanceId: 06795f2b
installDate: 7/6/2019 12:19:41 AM
installationName: VisualStudio/15.9.13+28307.718
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools
installationVersion: 15.9.28307.718
productId: Microsoft.VisualStudio.Product.BuildTools
productPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\LaunchDevCmd.bat
state: 4294967295
isComplete: 1
isLaunchable: 1
isPrerelease: 0
isRebootRequired: 0
displayName: Visual Studio Build Tools 2017
description: The Visual Studio Build Tools allows you to build native and managed MSBuild-based applications without requiring the Visual Studio IDE. There are options to install the Visual C++ compilers and libraries, MFC, ATL, and C++/CLI support.
channelId: VisualStudio.15.Release
channelUri: https://aka.ms/vs/15/release/channel
enginePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service
releaseNotes: https://go.microsoft.com/fwlink/?LinkId=660692#15.9.13
thirdPartyNotices: https://go.microsoft.com/fwlink/?LinkId=660708
updateDate: 2019-07-06T05:19:41.6331221Z
catalog_buildBranch: d15.9
catalog_buildVersion: 15.9.28307.718
catalog_id: VisualStudio/15.9.13+28307.718
catalog_localBuild: build-lab
catalog_manifestName: VisualStudio
catalog_manifestType: installer
catalog_productDisplayVersion: 15.9.13
catalog_productLine: Dev15
catalog_productLineVersion: 2017
catalog_productMilestone: RTW
catalog_productMilestoneIsPreRelease: False
catalog_productName: Visual Studio
catalog_productPatchVersion: 13
catalog_productPreReleaseMilestoneSuffix: 1.0
catalog_productRelease: RTW
catalog_productSemanticVersion: 15.9.13+28307.718
catalog_requiredEngineVersion: 1.18.1049.33485
properties_campaignId:
properties_channelManifestId: VisualStudio.15.Release/15.9.13+28307.718
properties_nickname:
properties_setupEngineFilePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installershell.exe

instanceId: 0ecf9214
installDate: 7/5/2019 6:02:28 PM
installationName: VisualStudio/16.1.5+29025.244
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools
installationVersion: 16.1.29025.244
productId: Microsoft.VisualStudio.Product.BuildTools
productPath: C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\LaunchDevCmd.bat
state: 4294967295
isComplete: 1
isLaunchable: 1
isPrerelease: 0
isRebootRequired: 0
displayName: Visual Studio Build Tools 2019
description: The Visual Studio Build Tools allows you to build native and managed MSBuild-based applications without requiring the Visual Studio IDE. There are options to install the Visual C++ compilers and libraries, MFC, ATL, and C++/CLI support.
channelId: VisualStudio.16.Release
channelUri: https://aka.ms/vs/16/release/channel
enginePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service
releaseNotes: https://go.microsoft.com/fwlink/?LinkId=660893#16.1.5
thirdPartyNotices: https://go.microsoft.com/fwlink/?LinkId=660909
updateDate: 2019-07-05T23:02:28.6552697Z
catalog_buildBranch: d16.1
catalog_buildVersion: 16.1.29025.244
catalog_id: VisualStudio/16.1.5+29025.244
catalog_localBuild: build-lab
catalog_manifestName: VisualStudio
catalog_manifestType: installer
catalog_productDisplayVersion: 16.1.5
catalog_productLine: Dev16
catalog_productLineVersion: 2019
catalog_productMilestone: RTW
catalog_productMilestoneIsPreRelease: False
catalog_productName: Visual Studio
catalog_productPatchVersion: 5
catalog_productPreReleaseMilestoneSuffix: 1.0
catalog_productSemanticVersion: 16.1.5+29025.244
catalog_requiredEngineVersion: 2.1.3096.46871
properties_campaignId:
properties_channelManifestId: VisualStudio.16.Release/16.1.5+29025.244
properties_nickname:
properties_setupEngineFilePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installershell.exe
```

Only `windows-build-tools`:
```
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationVersion
15.9.28307.718
$ vswhere.exe -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64
Visual Studio Locator version 2.5.2+gebb9f26a3d [query version 1.18.21.37008]
Copyright (C) Microsoft Corporation. All rights reserved.

instanceId: 5b6d4f0d
installDate: 7/6/2019 12:36:24 AM
installationName: VisualStudio/15.9.13+28307.718
installationPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools
installationVersion: 15.9.28307.718
productId: Microsoft.VisualStudio.Product.BuildTools
productPath: C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\Common7\Tools\LaunchDevCmd.bat
isPrerelease: 0
displayName: Visual Studio Build Tools 2017
description: The Visual Studio Build Tools allows you to build native and managed MSBuild-based applications without requiring the Visual Studio IDE. There are options to install the Visual C++ compilers and libraries, MFC, ATL, and C++/CLI support.
channelId: VisualStudio.15.Release
channelUri: https://aka.ms/vs/15/release/channel
enginePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\resources\app\ServiceHub\Services\Microsoft.VisualStudio.Setup.Service
releaseNotes: https://go.microsoft.com/fwlink/?LinkId=660692#15.9.13
thirdPartyNotices: https://go.microsoft.com/fwlink/?LinkId=660708
updateDate: 2019-07-06T07:36:24.0949931Z
catalog_buildBranch: d15.9
catalog_buildVersion: 15.9.28307.718
catalog_id: VisualStudio/15.9.13+28307.718
catalog_localBuild: build-lab
catalog_manifestName: VisualStudio
catalog_manifestType: installer
catalog_productDisplayVersion: 15.9.13
catalog_productLine: Dev15
catalog_productLineVersion: 2017
catalog_productMilestone: RTW
catalog_productMilestoneIsPreRelease: False
catalog_productName: Visual Studio
catalog_productPatchVersion: 13
catalog_productPreReleaseMilestoneSuffix: 1.0
catalog_productRelease: RTW
catalog_productSemanticVersion: 15.9.13+28307.718
catalog_requiredEngineVersion: 1.18.1049.33485
properties_campaignId:
properties_channelManifestId: VisualStudio.15.Release/15.9.13+28307.718
properties_nickname:
properties_setupEngineFilePath: C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installershell.exe
```

**Comment by unbornchikken** [comment]
Ok, thanks. I'm gonna release it as a minor patch, and looking forward for more feedback on this.

**Comment by unbornchikken** [comment]
Released as of v5.3.0.
